src/Equinor.ProCoSys.IPO.Query/GetInvitationsQueries/GetInvitationsForExport/GetInvitationsForExcelQueryHandler.cs

```diff
@@ -181,21 +181,27 @@ where personIds.Contains(p.Id)
         {
             var organizerIds = invitationsWithIncludes.Select(i => i.CreatedById).Distinct().ToList();
             var organizers = await GetPersonsByIdsAsync(organizerIds);
+            var organizersDict = organizers.ToDictionary(o => o.Id);
+
+            var invitationProjectIds = invitationsWithIncludes.Select(i => i.ProjectId).ToList();
+
+            var projects = await _context.QuerySet<Project>()
+                .Where(p => invitationProjectIds.Contains(p.Id))
+                .ToListAsync();
 
             var exportInvitations = new List<ExportInvitationDto>();
             foreach (var invitation in invitationsWithIncludes)
             {
-                var project = await _context.QuerySet<Project>()
-                    .SingleOrDefaultAsync(x => x.Id == invitation.ProjectId);
+                var project = projects.Single(x => x.Id == invitation.ProjectId);
 
                 if (project is null)
                 {
                     throw new ArgumentNullException(nameof(project));
                 }
 
-                var organizer = organizers.Single(x => x.Id == invitation.CreatedById);
+                var organizer = organizersDict[invitation.CreatedById];
                 var invitationWithIncludes = invitationsWithIncludes.Single(i => i.Id == invitation.Id);
-                var participants = invitationWithIncludes.Participants.ToList();
+                var participants = await Task.FromResult(invitationWithIncludes.Participants.ToList());
```

Comment:
Why this change?

Reply:
Good question. No need for (the attempted) async enumeration here since the collection has already been loaded at this stage. Reverting back to previous implementation.